src/python/pants/bin/goal_runner.py

```diff
@@ -241,3 +260,25 @@ def run(self):
         NailgunProcessGroup().killall()
 
     return result
+
+  def _execute_v2_rules(self):
+    self._context.graph_helper.run_console_rules(
+      self._context.requested_goals,
+      self._context.v2_target_roots
+    )
+    return 0
+
+  def run(self):
+    v1_results = v2_results = 0
+    global_options = self._context.options.for_global_scope()
+    self._validate_workdir(global_options.pants_workdir)
+
+    # N.B. For daemon runs, console rules execute pre-fork.
```

Comment:
I think that my thinking on this was that `@console_rules` always ran "first"... in the sense that they were sortof a different phase of the `PantsRunner` that didn't require the `GoalRunner` at all. Then the `GoalRunner` could (optionally) be created after they had run.

Reply:
makes sense. refactored this a bit - let me know what you think now.